darshan-runtime/lib/darshan-posix.c

```diff
@@ -1620,6 +1636,13 @@ int DARSHAN_DECL(close)(int fd)
             rec_ref->file_rec->fcounters[POSIX_F_META_TIME],
             tm1, tm2, rec_ref->last_meta_end);
         darshan_delete_record_ref(&(posix_runtime->fd_hash), &fd, sizeof(int));
+
+#ifdef HAVE_LDMS
+        /* publish close information for posix */
+        extern struct darshanConnector dC;
+        if(!dC.posix_enable_ldms)
```

Comment:
I noticed that in the close routine, you're not also checking the value of `dC.ldms_lib` like you do in open/read/write routines. This is true for all modules, not just POSIX.

You might want to add that check just to be safe. This dC struct is only partially initialized, so I think unset values are implicitly initialized to 0. Given my comments elsewhere about some of this logic being ambiguous/inverted, I think it might just be safest to always check the value is as expected?